pkg/gitops/repository.go

```diff
@@ -0,0 +1,180 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+//go:generate moq -out repository_moq_test.go . repositorier
+type repositorier interface {
+	Close(ctx context.Context) []error
+	localPath() string
+	gitClone() error
+	workingDirectoryClean() (bool, error)
+	gitCheckoutNewBranch() error
+	gitCommitAndPush(message string) error
+	openPullRequest(ctx context.Context, title, body string) (string, error)
+}
+
+// repository implements the repositorier interface.
+var _ repositorier = (*repository)(nil)
+
+type repository struct {
+	gh          githuber
+	remote      RemoteConfig
+	sshKey      sshKeyer
+	tmpRepoPath string
+}
+
+// RemoteConfig is a git remote configuration.
+type RemoteConfig struct {
+	URL, Branch, Folder string
+}
+
+// NewRepositoryParams are parameters for NewRepository function.
+type NewRepositoryParams struct {
+	Github githuber
```

Comment:
Those are interfaces which aren't necessary for the use of the package. I'm always trying to minimise API of a package to present only the relevant parts to it's user (only export what is actually used, interfaces can easily be re-declared because of duck typing in case they are needed)

Reply:
Yeah, but honestly I was a bit baffled as to how this even compiles, since `main.go` creates a `NewRepositoryParams`, it's so weird that these field are then not exported. But if this works, then I'm also all for hiding everything except what is absolutely required.